### PR TITLE
Use par for EVM actions

### PR DESCRIPTION
**Pull request by samholmes**
#### PR Dependencies

 none

#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [ ] Tested on iOS device
- [ ] Tested on Android device
- [ ] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202998266966430

### CODE REVIEW

File: src/util/ActionProgramUtils.ts [hunk 1]
```diff
@@ -72,7 +72,7 @@ export const makeAaveBorrowAction = async ({
   destBankId?: string
   nativeAmount: string
 }): Promise<ActionOp[]> => {
-  const out = []
+  const out: ActionOp[] = []
```
Comment (swansontec) [inline src/util/ActionProgramUtils.ts:132]: These TypeScript fixes make me happy. 👍

File: src/controllers/action-queue/types/pushCleaners.ts [hunk 2]
```diff
@@ -45,7 +58,16 @@ export const asTxConfirmTrigger: Cleaner<TxConfirmTrigger> = asObject({
...
+export const asPushTrigger: Cleaner<PushTrigger> = asEither(
+  asAddressBalanceTrigger,
+  asAllTrigger,
+  asAnyTrigger,
+  asPriceChangeTrigger,
+  asPriceLevelTrigger,
+  asTxConfirmTrigger
+)
+
+export const asPushTriggerState: Cleaner<PushTriggerState> = asOptional(asDate)
```
Comment (swansontec) [inline src/controllers/action-queue/types/pushCleaners.ts]: This type should be:

```ts
export const asPushTriggerState: Cleaner<PushTriggerState> = asOptional(
  asEither(
    asArray(raw => asPushTriggerState(raw)),
    asDate
  )
)
```

File: src/controllers/action-queue/types/pushTypes.ts [hunk 3]
```diff
@@ -34,7 +44,12 @@ export type TxConfirmTrigger = {
   readonly txid: string
 }
 
-export type PushTrigger = AddressBalanceTrigger | PriceChangeTrigger | PriceLevelTrigger | TxConfirmTrigger
+export type PushTrigger = AddressBalanceTrigger | AllTrigger | AnyTrigger | PriceChangeTrigger | PriceLevelTrigger | TxConfirmTrigger
+
+/**
+ * Records when a trigger took place.
+ */
+export type PushTriggerState = undefined | Date
```
Comment (swansontec) [inline src/controllers/action-queue/types/pushTypes.ts]: This type should be:

```ts
export type PushTriggerState =
  | undefined
  // For "any" and "all" triggers:
  | PushTriggerState[]
  // For normal triggers:
  | Date
```

**Review by swansontec — APPROVED** [review]
Approving, since I see that the type mistakes do get corrected later in the PR. It might make sense to put this last commit first in the series, so it starts out the right way around.

File: src/controllers/action-queue/types/pushTypes.ts (hunk 3 above)
Comment (samholmes) [reply src/controllers/action-queue/types/pushTypes.ts]: I fixed it in the first commit. I'm going to leave the last commit intact for reworking the files to be more exact with the push-server files

**Pull request merged**